### ISSUE TITLE
strange loss change

### ISSUE THREAD

**Issue by lixin7895123** [BVLC/caffe#1255]
I try to train a CNN with my model, and the loss changed in a weird way, the accuracy don't increase at all. I have no idea how could this happened? Is this a common issue?
As showed bellow, loss change from 87.3366 to 8.26466, in 1000 iteration, and accuracy never get better.

I1010 17:36:18.471521 24907 solver.cpp:160] Solving v1.2
I1010 17:36:18.471542 24907 solver.cpp:247] Iteration 0, Testing net (#0)
I1010 17:36:51.366374 24907 solver.cpp:298]     Test net output #0: accuracy = 0.00028
I1010 17:36:51.521034 24907 solver.cpp:191] Iteration 0, loss = 87.3366
I1010 17:36:51.521077 24907 solver.cpp:403] Iteration 0, lr = 0.001
I1010 17:40:30.682349 24907 solver.cpp:191] Iteration 1000, loss = 8.26466
I1010 17:40:30.682417 24907 solver.cpp:403] Iteration 1000, lr = 0.001
I1010 17:44:09.906486 24907 solver.cpp:191] Iteration 2000, loss = 8.25553
I1010 17:44:09.906579 24907 solver.cpp:403] Iteration 2000, lr = 0.001
I1010 17:47:49.071782 24907 solver.cpp:191] Iteration 3000, loss = 8.24047
I1010 17:47:49.071879 24907 solver.cpp:403] Iteration 3000, lr = 0.001
I1010 17:51:27.956398 24907 solver.cpp:247] Iteration 4000, Testing net (#0)
I1010 17:52:00.040813 24907 solver.cpp:298]     Test net output #0: accuracy = 0.00028
I1010 17:52:00.188971 24907 solver.cpp:191] Iteration 4000, loss = 8.28462
I1010 17:52:00.189007 24907 solver.cpp:403] Iteration 4000, lr = 0.001
I1010 17:55:39.479957 24907 solver.cpp:191] Iteration 5000, loss = 8.27163
I1010 17:55:39.480054 24907 solver.cpp:403] Iteration 5000, lr = 0.001
I1010 17:59:18.680330 24907 solver.cpp:191] Iteration 6000, loss = 8.24231
I1010 17:59:18.680424 24907 solver.cpp:403] Iteration 6000, lr = 0.001
I1010 18:02:57.882889 24907 solver.cpp:191] Iteration 7000, loss = 8.26924
I1010 18:02:57.882987 24907 solver.cpp:403] Iteration 7000, lr = 0.001
I1010 18:06:37.034945 24907 solver.cpp:317] Snapshotting to ./v1.2/v1.2_iter_iter_8000.caffemodel
I1010 18:06:37.067426 24907 solver.cpp:324] Snapshotting solver state to ./v1.2/v1.2_iter_iter_8000.solverstate
I1010 18:06:37.087664 24907 solver.cpp:247] Iteration 8000, Testing net (#0)
I1010 18:07:09.140758 24907 solver.cpp:298]     Test net output #0: accuracy = 0.00028
